nifi-nar-bundles/nifi-html-bundle/nifi-html-processors/src/main/java/org/apache/nifi/PutHTMLElement.java

```diff
@@ -0,0 +1,150 @@
+            .required(true)
+            .allowableValues(APPEND_ELEMENT, PREPEND_ELEMENT)
+            .defaultValue(APPEND_ELEMENT)
+            .build();
+
+    public static final PropertyDescriptor PUT_VALUE = new PropertyDescriptor.Builder()
+            .name("Put Value")
+            .description("Value used when creating the new Element. Value should be a valid HTML element. " +
+                    "The text should be supplied unencoded: characters like '<', '>', etc will be properly HTML " +
+                    "encoded in the output.")
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(true)
+            .build();
+
+    private List<PropertyDescriptor> descriptors;
+
+    private Set<Relationship> relationships;
+
+    @Override
+    protected void init(final ProcessorInitializationContext context) {
+        final List<PropertyDescriptor> descriptors = new ArrayList<PropertyDescriptor>();
+        descriptors.add(URL);
+        descriptors.add(CSS_SELECTOR);
+        descriptors.add(HTML_CHARSET);
+        descriptors.add(PUT_LOCATION_TYPE);
+        descriptors.add(PUT_VALUE);
+        this.descriptors = Collections.unmodifiableList(descriptors);
+
+        final Set<Relationship> relationships = new HashSet<Relationship>();
+        relationships.add(REL_ORIGINAL);
+        relationships.add(REL_SUCCESS);
+        relationships.add(REL_FAILURE);
+        relationships.add(REL_INVALID_HTML);
+        this.relationships = Collections.unmodifiableSet(relationships);
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return this.relationships;
+    }
+
+    @Override
+    public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return descriptors;
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        final FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            return;
+        }
+
+        try {
+            final Document doc = parseHTMLDocumentFromFlowfile(flowFile, context, session);
+            final Elements eles = doc.select(context.getProperty(CSS_SELECTOR).evaluateAttributeExpressions().getValue());
+
+            if (eles == null || eles.size() == 0) {
+                //No element found
+                session.transfer(flowFile, REL_NOT_FOUND);
+            } else {
+                for (Element ele : eles) {
+                    switch (context.getProperty(PUT_LOCATION_TYPE).getValue()) {
+                        case APPEND_ELEMENT:
+                            ele.append(context.getProperty(PUT_VALUE).evaluateAttributeExpressions().getValue());
+                            break;
+                        case PREPEND_ELEMENT:
+                            ele.prepend(context.getProperty(PUT_VALUE).evaluateAttributeExpressions().getValue());
```

Comment:
This should call evaluateAttributeExpressions(flowFile) so that we can take into account FlowFile Attributes.